blog/2025/pcsx2-2.6/index.mdx

```diff
@@ -0,0 +1,284 @@
+This required us to either hardcode or generate ranges for various languages.
+
+With 1.9.21, we can now rely on ImGui to dynamically adjust to font atlas, removing the maintenance burden of fixed character ranges.
+
+1.9.21 also provides us an option to disable some of ImGui's navigation system, allowing us to remove a patch we had been applying before.
+
+Before:
+
+<Image src={require("./img/achievement_emoji_before.webp").default} />
+
+After:
+
+<Image src={require("./img/achievement_emoji_after.webp").default} />
+
+## Quality of Life Improvements
+
+<PCSX2PRLink authors="TheTechnician27" prNums="13388">
+  ### Add Qt Buddies and Tabstops
+</PCSX2PRLink>
+
+This addition helps with screen reading Accessibility and how settings in PCSX2 is presented within PCSX2. [Buddy](https://doc.qt.io/qt-6.8/designer-buddy-mode.html) and [Tabstops](https://doc.qt.io/qt-6.8/designer-tab-order.html) are a very useful feature provided by Qt that allows you to build a more accessible user interface.
+
+#### Buddies
+
+Buddies are a property in Qt which associate a label with an interactive element. This is transparent for most users, but for users with a screen reader, this means you can actually tell what an interactive element does. It's a win for accessibility with minimal maintenance overhead and arguably better self-documentation.
+
+Currently, for example, focusing a dropdown will just say that you've focused a dropdown without saying what it's supposed to be for – rendering much of our UI totally unusable. With a buddy, it will read out the label on top of the element type. As an example, with buddies, the OSD scale option will read approximately as "OSD Scale: Spin Box 100", but without, it will only read "Spin Box 100". There is no way to focus the labels either, so it's unlikely they can be read out without buddies.
+
+<Image src={require("./img/buddy.webp").default} />
+
+#### Tabstops
+
+Have you been in a situation where you have to use a keyboard to navigate your way through PCSX2 and found out that the ordering between options are a complete nonsense? Tabstops is here to help!
+
+Tabstops are another property of Qt which allows an explicit ordering of elements when tabbing through a widget (i.e. focusing on one element then pressing Tab). Without them, the tab order is subject to the physical ordering of the elements in XML, which does not inherently correspond to the row/column order displayed and is highly unreliable and unrealistic to maintain.
+
+<Image src={require("./img/tabstops.webp").default} />
+
+<PCSX2PRLink authors="TellowKrinkle" prNums="12829,12830">
+  ### Improvements to PCSX2 related files handling on macOS
+</PCSX2PRLink>
+
+TellowKrinkle, our resident macOS housekeeper, has been working on adding file association to some of file that can be opened by PCSX2 such as game dumps (ISO, CSO, ZSO, CHD, etc), save states, GS Dumps, PNACH Patches, etc.
```

Comment:
"has been working on adding file type associations for some of the file types that can be opened by PCSX2. This includes game dumps (ISO, CSO, ZSO, CHD, etc), save states, GS dumps, PNACH patches, and more."